js/components/graph/EdgeGroup.js

```diff
@@ -19,28 +15,20 @@ export default class EdgeGroup extends React.Component {
    * @param {string} edgeID
    * @param {string} state
    */
-  updateEdgeStatus = (edgeID, state) => {
+  updateEdgeMissingStatus = (edgeID, state) => {
     var isMissing = state === "missing"
-    this.setState({ [edgeID]: isMissing })
-  }
-
-  /**
-   * After each render after the initial, update the status of each Edge
-   */
-  componentDidUpdate() {
-    this.props.edgesJSON.forEach(edgeJSON => {
-      this[edgeJSON.id_].updateStatus()
-    })
+    this.props.updateEdgeMissingStatus(edgeID, isMissing)
   }
 
   /**
    * Set all Edges to the inactive status
    */
   reset = () => {
```

Comment:
Instead of modifying this method, you should move the corresponding logic into the `Graph.reset` method so that it can reset the edge status state itself (rather than this indirect approach of `Graph.reset` calling `EdgeGroup.reset`.

This will also allow you to delete the `ref` that the `Graph` has to the `EdgeGroup` child component.